packages/cursorless-engine/src/processTargets/targets/BaseTarget.ts

```diff
@@ -30,6 +30,7 @@ export interface CloneWithParameters {
 
 export default abstract class BaseTarget implements Target {
   protected readonly state: CommonTargetParameters;
+  isDocument = false;
```

Comment:
I had to add this to handle the case where we are in a document that only contains a class.  In that case we want to prefer the class method function snippet definition even though it's the same range as the whole document.  We could probably remove this if we add a mechanism for the class method snippet to deactivate itself if there's a function in the ancestor chain between it and its containing class